File: java_spring_hw/src/main/java/com/example/java_spring_hw/controller/TaskController.java
package com.example.java_spring_hw.controller;

import com.example.java_spring_hw.entity.Task;
import com.example.java_spring_hw.request.TaskRequest;
import com.example.java_spring_hw.service.TaskService;
import lombok.AllArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@AllArgsConstructor
@RequestMapping("/api/tasks")
@RestController
public class TaskController {
    private final TaskService taskService;

    @PostMapping
    public Task create(@RequestBody TaskRequest request) {
        return taskService.create(request);
    }

    @GetMapping("/{id}")
    public Task get(@PathVariable long id) {
        return taskService.get(id);
    }

    @GetMapping
    public List<Task> getAll() {
        return taskService.getAll();
    }

    @GetMapping("/{id}/start")
    public void startTask(@PathVariable long id) {
        taskService.startTask(id);
    }
}

File: kotlin_quarkus_hw/src/main/kotlin/com/example/controller/TaskController.kt
package com.example.controller

import com.example.entity.Task
import com.example.request.TaskRequest
import com.example.service.TaskService
import javax.ws.rs.*

@Path("/api/tasks")
class TaskController(
    private val taskService: TaskService
) {

    @POST
    fun create(request: TaskRequest): Task {
        return taskService.create(request)
    }

    @GET
    @Path("/{id}")
    fun getById(@PathParam("id") id: Long): Task {
        return taskService.get(id)
    }

    @GET
    fun getAll(): List<Task> {
        return taskService.getAll()
    }

}